cmd/crank/pkg/lint.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pkg
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/stdcopy"
+
+	"github.com/crossplane/crossplane-runtime/pkg/parser"
+	"github.com/crossplane/crossplane/pkg/controller/pkg/revision"
+)
+
+// LintCmd uses the crossplane package controller linter to validate the contents of a package.
+// The rules for linting different kinds of packages (configuration, provider) are determined
+// by the package type's subcommand, so the *revision.PackageLinter is passed in by the calling cmd
+type LintCmd struct {
+	Image  string `arg:"" optional:"" name:"image" help:"Name of the package image to run and lint the output of."`
+	NoPull bool   `help:"Flag to disable pulling the package image before running it."`
+}
```

Comment:
It feels a little weird to model this as a `kong` command since it can't actually be invoked by the user, only by other commands (e.g. will the `help` ever be exposed here?), but it seems to be working as intended, so likely okay to move forward as-is for now :+1: